src/pages/FarmPage/FarmsList.tsx

```diff
@@ -267,67 +261,42 @@ const FarmsList: React.FC<FarmsListProps> = ({ bulkPairs, farmIndex }) => {
         .reduce((totStr, str) => totStr + str, '')
     : null;
 
-  useEffect(() => {
-    setStakingInfos(undefined);
-    setStakingDualInfos(undefined);
-    setTimeout(() => setPageLoading(false), 500); //load farms 0.5s after loading page
-  }, []);
-
   useEffect(() => {
     setPageIndex(0);
-    if (farmIndex === GlobalConst.farmIndex.LPFARM_INDEX) {
-      setStakingInfos(sortedLPStakingInfos.slice(0, LOADFARM_COUNT));
-    } else {
-      setStakingDualInfos(sortedStakingDualInfos.slice(0, LOADFARM_COUNT));
-    }
-    return () => {
-      setStakingInfos(undefined);
-      setStakingDualInfos(undefined);
-    };
-    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [stakingRewardAddress]);
 
-  useEffect(() => {
-    if (farmIndex === GlobalConst.farmIndex.LPFARM_INDEX) {
-      const currentStakingInfos = stakingInfos || [];
-      const stakingInfosToAdd = sortedLPStakingInfos.slice(
-        currentStakingInfos.length,
-        currentStakingInfos.length + LOADFARM_COUNT,
-      );
-      setStakingInfos(currentStakingInfos.concat(stakingInfosToAdd));
-    } else if (farmIndex === GlobalConst.farmIndex.DUALFARM_INDEX) {
-      const currentDualStakingInfos = stakingDualInfos || [];
-      const stakingDualInfosToAdd = sortedStakingDualInfos.slice(
-        currentDualStakingInfos.length,
-        currentDualStakingInfos.length + LOADFARM_COUNT,
-      );
-      setStakingDualInfos(
-        currentDualStakingInfos.concat(stakingDualInfosToAdd),
-      );
+  const stakingInfos = useMemo(() => {
+    return sortedLPStakingInfos
+      ? sortedLPStakingInfos.slice(
+          0,
+          pageIndex === 0 ? LOADFARM_COUNT : LOADFARM_COUNT * pageIndex,
```

Comment:
No need for ternary `(LOADFARM_COUNT * pageIndex) + LOADFARM_COUNT `

Reply:
This should probably be a function on an array called  `getPage(index, count)`

Reply:
fixed this by creating a reusable functions `getPageItemsToLoad(index: number, countsPerPage: number)`

Reply:
I don't see you calling the function you made did you forget to push?